samples/apps/copilot-chat-app/webapi/Skills/DocumentQuerySkill.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Globalization;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SemanticFunctions.Partitioning;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace SemanticKernel.Service.Skills;
+
+/// <summary>
+///
+/// </summary>
+public class DocumentQuerySkill
+{
+    /// <summary>
+    /// Returns the name of the semantic document memory collection that stores document semantic memory.
+    /// </summary>
+    /// <param name="userId">ID of the user who owns the documents.</param>
+    internal string UserDocumentMemoryCollectionName(string userId) => $"{userId}-documents";
+
+    /// <summary>
+    /// Name of the semantic document memory collection that stores document semantic memory globally
+    /// available to users accessing the service.
+    /// </summary>
+    internal string GlobalDocumentMemoryCollectionName = "global-documents";
+
+    /// <summary>
+    /// Prompt settings.
+    /// </summary>
+    private readonly PromptSettings _promptSettings;
+
+    /// <summary>
+    /// Create a new instance of DocumentQuerySkill.
+    /// </summary>
+    public DocumentQuerySkill(PromptSettings promptSettings)
+    {
+        this._promptSettings = promptSettings;
+    }
+
+    /// <summary>
+    /// Parse a local file into embeddings.
+    /// </summary>
+    /// <param name="localFile">Path to the local file including the file name.</param>
+    /// <param name="context">Contains the 'audience' indicating the name of the user.</param>
+    [SKFunction("Parse a local file")]
+    [SKFunctionName("ParseLocalFile")]
+    [SKFunctionInput(Description = "Path to the local file including the file name.")]
+    [SKFunctionContextParameter(
+        Name = "userId",
+        Description = "ID of the user who owns the documents. This is used to create a unique collection name for the user." +
+                      "If the user ID is not specified or empty, the documents will be stored in a global collection.",
+        DefaultValue = "")]
+    public async Task ParseLocalFileAsync(string localFile, SKContext context)
+    {
+        string collection = context.Variables.ContainsKey("userID") ?
+            string.IsNullOrEmpty(context.Variables["userID"]) ?
+                GlobalDocumentMemoryCollectionName :
+                UserDocumentMemoryCollectionName(context.Variables["userID"]) :
+            GlobalDocumentMemoryCollectionName;
+
+        string text = string.Empty;
+        try
+        {
+            text = File.ReadAllText(localFile);
```

Comment:
You can use ReadAllTextAsync() instead:

```suggestion
            text = await File.ReadAllTextAsync(localFile);
```